src/ConsoleArgumentBag.php

```diff
@@ -80,4 +82,22 @@ public function getCommandName(): string
     {
         return $this->path[1] ?? '';
     }
+
+    private function castArgument(ConsoleInputArgument $argument, ConsoleArgumentDefinition $argumentDefinition): ConsoleInputArgument
```

Comment:
This should (eventually?) live in the core (https://github.com/tempestphp/tempest-framework/pull/229 was my initial attempt at solving that, requires quite a bit of work, I don't like the API anymore like at all lol, lemme know if that's something you'd want me to revisit

Reply:
dropped enum support from this, will add contribute that to core instead

Reply:
I did just add mapper support into tempest/core, so that might help?